spec/models/user_spec.rb

```diff
@@ -0,0 +1,123 @@
+      it "does not return users that don't start with the provided letter" do
+        User.find_mathing_letter("j").should_not include @tom
+      end
+    end
+  end
+
+  describe "handle callback function return" do
+  	it "google returned expected parameters" do
+  		jsonResponseString = {"provider"=>"google_oauth2",
+							 "uid"=>"103648144954862111999",
+							 "info"=>
+							  {"name"=>"Mantas Savickis",
+							   "email"=>"msdntas.ssdfvickis789582@gmail.com",
+							   "first_name"=>"Mantas",
+							   "last_name"=>"Savickis",
+							   "image"=>"https://lh4.googleusercontent.com/-xoTS5TunNgs/AAAAAAAAAAI/AAAAAAAAAAc/i3YMFT0FV9A/photo.jpg"},
+							 "credentials"=>
+							  {"token"=>"ya29.GlyXBbmAorp48h2TAgAVI0TmFfRcpUep0eUhxReD8ZWOA29OsT0eyBZAQSKDuIyp830hhGuS9BFhn509lcEJFVXu3rJe6ijsws_rJq1hR2auduBJo8Y-nuyzFzB_Nw", "expires_at"=>1523216443, "expires"=>true},
+							 "extra"=>
+							  {"id_token"=>
+							    "eyJhbGciOiJSUzI1NiIsImtpZCI6IjNiNTQ3ODg2ZmY4NWEzNDI4ZGY0ZjYxZGI3M2MxYzIzOTgyYTkyOGUifQ.eyJhenAiOiIzOTgxMzU1NDk3MjUtcG05c2hqaXU0ZnRhMjQwbmlyZGJiOGJhaWhha2Y0cGguYXBwcy5nb29nbGV1c2VyY29udGVudC5jb20iLCJhdWQiOiIzOTgxMzU1NDk3MjUtcG05c2hqaXU0ZnRhMjQwbmlyZGJiOGJhaWhha2Y0cGguYXBwcy5nb29nbGV1c2VyY29udGVudC5jb20iLCJzdWIiOiIxMDM2NDgxODI5NTE5NjIxMTE5OTkiLCJlbWFpbCI6Im1hbnRhcy5zYXZpY2tpczc4OTJAZ21haWwuY29tIiwiZW1haWxfdmVyaWZpZWQiOnRydWUsImF0X2hhc2giOiI4bzY4cnZoekZGMTlKdzNBWkYzOTFnIiwiZXhwIjoxNTIzMjE2NDQ0LCJpc3MiOiJhY2NvdW50cy5nb29nbGUuY29tIiwiaWF0IjoxNTIzMjEyODQ0fQ.Z9ZZ38Hr9TnsTlQP7fmYy2IcUVtPQO6OgutGAhtKDuhvFMzPNgOg_vBSwbIktESZfTe0_lmMVk_7ZEPSnIOJoeR1bZH6rTX0TrLuTIHcsidahrPuLtTiqGxYfGeO9hKkcotCa7_RwTTHz6ZjwYe_9NRau0lXBvTQ2bZprnk1s0Z7RQfoEISemReswhnwpODUetYKWXrK8Dj3CacwJUTvK9f0OlMEXcoMNg8MwJXaVR0qRxgsxDd8mzw1bi4b3iqRBr-fQYHk72F_3iJRMq4zWWpZOr86xvq04mJPzvNxVom7IQ_Q4HSN0W1snr-1RdQN0eKaUqi7dZZ1AvnshrIr8Q",
+							   "id_info"=>
+							    {"azp"=>"398135549725-pm9shjiu4fta240nirdbb8baihakf4ph.apps.googleusercontent.com",
+							     "aud"=>"398135549725-pm9shjiu4fta240nirdbb8baihakf4ph.apps.googleusercontent.com",
+							     "sub"=>"103648182951962111999",
+							     "email"=>"mantas.savickis7892@gmail.com",
+							     "email_verified"=>true,
+							     "at_hash"=>"8o68rvhzFF19Jw3AZF391g",
+							     "exp"=>1523216444,
+							     "iss"=>"accounts.google.com",
+							     "iat"=>1523212844},
+							   "raw_info"=>
+							    {"kind"=>"plus#personOpenIdConnect",
+							     "sub"=>"103648182951962111999",
+							     "name"=>"Mantas Savickis",
+							     "given_name"=>"Mantas",
+							     "family_name"=>"Savickis",
+							     "picture"=>"https://lh4.googleusercontent.com/-xoTS5TunNgs/AAAAAAAAAAI/AAAAAAAAAAc/i3YMFT0FV9A/photo.jpg?sz=50",
+							     "email"=>"mantas.savickis7892@gmail.com",
+							     "email_verified"=>"true",
+							     "locale"=>"en"}}}.to_json
+  	
+  	  parsedJson = JSON.parse(jsonResponseString, object_class: OpenStruct)
+  	  expect(User.from_omniauth(parsedJson)).not_to be_falsey
```

Comment:
you could use `double` here and reduce test setup dramaticly, For example:
```
auth_object = double(
      provider: 'google', 
      info: double(email: user.email),
      # ... more methods which are used inside from_omniauth
    )

    expect(User.from_omniauth(auth_object)).not_to be_falsey
```

Reply:
fixed, removed instance variable and replaced with double